### PR TITLE
DOCSP-43309: Initial 2.3.1 release notes

### PR DESCRIPTION
**Pull request by sarah-olson-mongodb**
## DESCRIPTION
Mongosh 2.3.1 Release Notes

## STAGING
https://deploy-preview-356--docs-mongodb-shell.netlify.app/changelog/

## JIRA
https://jira.mongodb.org/browse/DOCSP-43309

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/changelog.txt [hunk 1]
```diff
@@ -12,6 +12,18 @@ Release Notes
...
    :class: singlecol
 
+v2.3.1
+------
+
+*Released September 5, 2024*
+
+Bug fixes in this release:
+
+- :issue:`COMPASS-8252` - Remove certificates without issuer from the TLS CA list
```
Comment (jmd-mongo) [inline source/changelog.txt:22]: [question]

Should we also list [MONGOSH-1416](https://jira.mongodb.org/browse/MONGOSH-1416) and [MONGOSH-1859](https://jira.mongodb.org/browse/MONGOSH-1859) here? Thank you!

**Review by jmd-mongo — COMMENTED** [review]
Thanks for these release notes, @sarah-olson-mongodb! I've left a question for your consideration.

Thanks,
Joe

File: source/changelog.txt (hunk 1 above)
Comment (sarah-olson-mongodb) [reply source/changelog.txt:22]: Thanks @jmd-mongo! Great question. I'm basing this off of @addaleax's release summary in the [mongosh](https://mongodb.slack.com/archives/CUHC9R8J0/p1725567397992589) slack channel.

**Review by jmd-mongo — APPROVED** [review]
Thanks, @sarah-olson-mongodb! This one LGTM.

Thank you,
Joe

File: source/changelog.txt (hunk 1 above)
Comment (addaleax) [reply source/changelog.txt:22]: I think we can list MONGOSH-1859 here, yeah, as e.g. "`ISODate()` now passes non-string arguments to `new Date()`". 🙂

Comment (sarah-olson-mongodb) [reply source/changelog.txt:22]: Thanks both and updated!

**Pull request merged**